pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go

```diff
@@ -289,6 +319,32 @@ func (m *managerImpl) processShutdownEvent() error {
 	klog.V(1).InfoS("Shutdown manager processing shutdown event")
 	activePods := m.getPods()
 
+	defer func() {
+		m.dbusCon.ReleaseInhibitLock(m.inhibitLock)
+		klog.V(1).InfoS("Shutdown manager completed processing shutdown event, node will shutdown shortly")
+	}()
+
+	if m.enableMetrics && m.storage != nil {
+		startTime := time.Now()
+		err := m.storage.Store(state{
+			StartTime: startTime,
+		})
+		if err != nil {
+			klog.ErrorS(err, "Failed to store graceful shutdown state")
+		}
+
+		defer func() {
+			endTime := time.Now()
+			err := m.storage.Store(state{
+				StartTime: startTime,
```

Comment:
I think we should emit the start time when we start processing and also try to defer emit the end time. In the worst case, the end time is never emitted and we can find such nodes from metrics data.

Reply:
+1 makes sense, if there is limited time to shutdown, it's hard to guarantee we will have enough time to cleanly release inhibit lock.

Reply:
> I think we should emit the start time when we start processing and also try to defer emit the end time. In the worst case, the end time is never emitted and we can find such nodes from metrics data.

Yes, the previous is to record the start time separately, here is to record the start and end time at the same time after completion.


> +1 makes sense, if there is limited time to shutdown, it's hard to guarantee we will have enough time to cleanly release inhibit lock.

It seems that extending the inhibit lock time a little bit can alleviate this situation.

Reply:
>  Yes, the previous is to record the start time separately, here is to record the start and end time at the same time after completion.
Can we also set `GracefulShutdownStartTime` there? So, in case the node never comes back up, we have a chance to get the start time.